piccolo_api/jwt_auth/middleware.py

```diff
@@ -137,9 +137,15 @@ async def __call__(self, scope, receive, send):
         """
         allow_unauthenticated = self.allow_unauthenticated
 
-        if scope["path"] in self.excluded_paths:
-            await self.asgi(scope, receive, send)
-            return
+        for excluded_path in self.excluded_paths:
+            if excluded_path.endswith("*"):
+                if excluded_path.startswith(excluded_path.rstrip("*")):
```

Comment:
This isn't quite right - because it's comparing `excluded_paths` with `excluded_paths`.

Reply:
But 
```python
if scope['path'].startswith(excluded_path.rstrip('*')):
```
does nothing.  If we have `/foo/*` we need check to `root_path` like this 
```python
if scope["root_path"].startswith(excluded_path.rstrip("/*")): 
```
to take some effect. Sorry if I don't understand well.

Reply:
```python
if scope['path'].startswith(excluded_path.rstrip('*')):
```
work only if we specified `excluded_paths=["*"]` not `excluded_paths=["/foo/*"]`. Did you think so?

Reply:
I think the confusion is to do with out FastAPI splits the path between `path` and `root_path`.

For example, if I have the an `APIRouter` mounted at `/private`, and an endpoint mounted to that router at `/blog`, then `root_path` is `/private` and `path` is `blog`.

I think we need to combine them for it to work properly. So rather than just checking against `path` it's something like `urllib.parse.urljoin(request.scope['root_path'], request.scope['path'])`.

Reply:
There's also `raw_path`:

https://asgi.readthedocs.io/en/latest/specs/www.html#http-connection-scope

Reply:
This discussion gives some more context https://github.com/django/asgiref/issues/229

Reply:
`raw_path` did the trick. If we use `raw_path` like this
```python
if scope["raw_path"].decode("utf-8").startswith(excluded_path.rstrip("*")):
```
both `excluded_paths=["*"]` and `excluded_paths=["/foo/*"]` works.